pkg/remediator/reconcile/worker_test.go

```diff
@@ -44,6 +45,184 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// TestWorker_Run_Remediates verifies that worker.Run remediates declared
+// objects added to the queue.
+func TestWorker_Run_Remediates(t *testing.T) {
+	ctx := context.Background()
+
+	existingObjs := []client.Object{
+		fake.ClusterRoleBindingObject(syncertest.ManagementEnabled),
+		fake.ClusterRoleObject(syncertest.ManagementEnabled),
+	}
+	declaredObjs := []client.Object{
+		fake.ClusterRoleBindingObject(syncertest.ManagementEnabled),
+		fake.ClusterRoleObject(syncertest.ManagementEnabled),
+	}
+	changedObjs := []client.Object{
+		queue.MarkDeleted(ctx, fake.ClusterRoleBindingObject()),
+		fake.ClusterRoleObject(syncertest.ManagementEnabled,
+			core.Label("new", "label")),
+	}
+	expectedObjs := []client.Object{
+		// CRB delete should be reverted
+		// TODO: Upgrade FakeClient to increment UID after deletion
+		fake.ClusterRoleBindingObject(syncertest.ManagementEnabled,
+			core.UID("1"), core.ResourceVersion("1"), core.Generation(1),
+		),
+		// Role change should be reverted
+		fake.ClusterRoleObject(syncertest.ManagementEnabled,
+			core.UID("1"), core.ResourceVersion("3"), core.Generation(1),
+		),
+	}
+
+	q := queue.New("test")
+	defer q.ShutDown()
+
+	c := testingfake.NewClient(t, core.Scheme)
```

Comment:
nit: use more descriptive variable names. (had to go back up here to find out what `c` was)

Reply:
This code is copied from the existing tests. Do you want me to refactor them all in this PR?

Reply:
no that's not necessary

Reply:
Variable naming is a bit of a holy war. I also prefer more descriptive names, but some go authors have said they prefer short ones. `c` in particular gets used a lot in the K8s libs, because `client` is a common package name.

If you feel strongly, I suggest doing some refactoring afterwards. I'd be happy to approve.